server/src/main/java/connection/ServiceManager.java

```diff
@@ -0,0 +1,23 @@
+package connection;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class ServiceManager {
+    private Map<String, GameCommand> map;
+
+    ServiceManager(Server server) {
+        this.map = new HashMap<>();
+        map.put("ADD_PLAYER", new AddPlayerGameCommand());
+        map.put("ASK_FOR_FLEET", new AskForFleetGameCommand(server));
+        map.put("CLIENT_READY", new ClientReadyGameCommand(server));
+        map.put("SHOT", new ClientReadyGameCommand(server));
+    }
+
+    GameCommand getCommand(String message) {
+        if (message.startsWith("ADD_PLAYER")) {
+            return map.get("ADD_PLAYER");
```

Comment:
message.startsWith("ADD_PLAYER") - but still you are doing:
map.get("ADD_PLAYER");

so it looks like this argument is never used

Reply:
It is used, look at MessageFromClientHandler and execute() method